bika/lims/validators.py

```diff
@@ -203,15 +206,17 @@ class InvoiceBatch_EndDate_Validator:
     name = "invoicebatch_EndDate_validator"
 
     def __call__(self, value, *args, **kwargs):
-        instance = kwargs['instance']
-        startdate = instance.getBatchStartDate()
-        # request = kwargs.get('REQUEST', {})
-        # form = request.get('form', {})
-        enddate = value
-        startdate = startdate.strftime('%Y-%m-%d %H:%M')
+        instance = kwargs.get('instance')
+        request = kwargs.get('REQUEST')
 
-        translate = getToolByName(instance, 'translation_service').translate
+        if request and request.form.get('BatchStartDate'):
+            startdate = _strptime(request.form.get('BatchStartDate'), '%Y-%m-%d %H:%M')
+        else:
+            startdate = _strptime(instance.getBatchStartDate(), '%Y-%m-%d %H:%M')
+
+        enddate = _strptime(value, '%Y-%m-%d %H:%M')
 
+        translate = senaiteapi.get_tool('translation_service', instance).translate
```

Comment:
Umm... there is no need to import `senaite.api `add-on imo. Just do `api.get_tool('translation_service', instance)` and remove the import from line 20.